worker-taskstowing/src/main/java/com/microfocus/caf/worker/taskstowing/factory/TaskStowingWorkerFactory.java

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright 2021 Micro Focus or one of its affiliates.
+ *
+ * The only warranties for products and services of Micro Focus and its
+ * affiliates and licensors ("Micro Focus") are set forth in the express
+ * warranty statements accompanying such products and services. Nothing
+ * herein should be construed as constituting an additional warranty.
+ * Micro Focus shall not be liable for technical or editorial errors or
+ * omissions contained herein. The information contained herein is subject
+ * to change without notice.
+ *
+ * Contains Confidential Information. Except as specifically indicated
+ * otherwise, a valid license is required for possession, use or copying.
+ * Consistent with FAR 12.211 and 12.212, Commercial Computer Software,
+ * Computer Software Documentation, and Technical Data for Commercial
+ * Items are licensed to the U.S. Government under vendor's standard
+ * commercial license.
+ */
+package com.microfocus.caf.worker.taskstowing.factory;
+
+import com.hpe.caf.api.Codec;
+import com.hpe.caf.api.ConfigurationException;
+import com.hpe.caf.api.ConfigurationSource;
+import com.hpe.caf.api.HealthResult;
+import com.hpe.caf.api.HealthStatus;
+import com.hpe.caf.api.worker.BulkWorker;
+import com.hpe.caf.api.worker.BulkWorkerRuntime;
+import com.hpe.caf.api.worker.DataStore;
+import com.hpe.caf.api.worker.InvalidTaskException;
+import com.hpe.caf.api.worker.JobStatus;
+import com.hpe.caf.api.worker.NotIndendedTaskMessageForwardingEvaluator;
+import com.hpe.caf.api.worker.TaskForwardingAction;
+import com.hpe.caf.api.worker.TaskInformation;
+import com.hpe.caf.api.worker.TaskMessage;
+import com.hpe.caf.api.worker.TaskRejectedException;
+import com.hpe.caf.api.worker.Worker;
+import com.hpe.caf.api.worker.WorkerCallback;
+import com.hpe.caf.api.worker.WorkerException;
+import com.hpe.caf.api.worker.WorkerFactory;
+import com.hpe.caf.api.worker.WorkerTask;
+import com.hpe.caf.api.worker.WorkerTaskData;
+import com.microfocus.caf.worker.taskstowing.TaskStowingBulkWorker;
+import com.microfocus.caf.worker.taskstowing.TaskStowingWorker;
+import com.microfocus.caf.worker.taskstowing.database.DatabaseClient;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public final class TaskStowingWorkerFactory implements WorkerFactory, NotIndendedTaskMessageForwardingEvaluator, BulkWorker
```

Comment:
I wasn't sure if this class needed to implement `WorkerFactory` as well as `BulkWorker`? The Job Tracking Worker does and the comments for the `BulkWorker` say:

```
This interface should be implemented by CAF Workers which are able to process multiple tasks together. It is expected to be implemented by the WorkerFactory object which has been supplied by the getWorkerFactory() method of the WorkerFactoryProvider class.
```

so I assume we have to implement both `WorkerFactory` as well as `BulkWorker`. I'm just unsure in what scenario `getWorker` will get called - in my testing I only saw it getting called if the `TaskStowingBulkWorker` failed to set a response on the task.

Reply:
I must have missed these comments yesterday.  You're right if would not get called at all in normal circumstances - only if there was some sort of error with the bulk infrastructure and it fell back to the sequential infrastructure.